java-snapshot-testing-core/src/main/java/au/com/origin/snapshots/reporters/SnapshotDiffReporter.java

```diff
@@ -0,0 +1,11 @@
+package au.com.origin.snapshots.reporters;
+
+import au.com.origin.snapshots.comparators.CompareResult;
+import au.com.origin.snapshots.comparators.SnapshotComparator;
+
+public interface SnapshotDiffReporter {
+
+    boolean supportsComparator(SnapshotComparator<?> comparator);
```

Comment:
Makes sense. 

But I'm on the fence about the enum. Won't it be too restrictive ? What if users have two additional custom output formats they want to use ? Won't 'OTHER' be too ambiguous in that case ? I feel it would be better to change supportsComparator to supportsSerializer. What do you think ?

Reply:
Or maybe a string serializer type ?

Reply:
Let's go with the string. Internally we can do SerializerType.JSON.name()